cmd/preview/cmdpreview.go

```diff
@@ -12,11 +12,14 @@ import (
 	cmdutil "k8s.io/kubectl/pkg/cmd/util"
 	"k8s.io/kubectl/pkg/util/i18n"
 	"sigs.k8s.io/cli-utils/pkg/apply"
+	"sigs.k8s.io/cli-utils/pkg/apply/event"
 )
 
 // NewCmdApply creates the `apply` command
 func NewCmdPreview(f util.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
 	applier := apply.NewApplier(f, ioStreams)
+	destroyer := apply.NewDestroyer(f, ioStreams)
```

Comment:
Could we somehow postpone creating these until we know which one we need? Maybe we can just check the flag in the `Run` function and then have a separate function for using the destroyer and the applier?